src/status_im2/contexts/chat/messages/link_preview/events.cljs

```diff
@@ -41,32 +42,96 @@
-                    :params     [community-id]
-                    :on-success #(rf/dispatch [:chat.ui/community-resolved community-id %])
-                    :on-error   #(do
-                                   (rf/dispatch [:chat.ui/community-failed-to-resolve community-id])
-                                   (log/error "Failed to request community info from mailserver"))}]})
+(rf/reg-event-fx :chat.ui/community-failed-to-resolve community-failed-to-resolve)
+
+(defn fetch-community
+  [{:keys [db]} [community-id]]
+  (when community-id
+    {:db            (assoc-in db [:communities/resolve-community-info community-id] true)
+     :json-rpc/call [{:method     "wakuext_fetchCommunity"
+                      :params     [{:CommunityKey    community-id
+                                    :TryDatabase     true
+                                    :WaitForResponse true}]
+                      :on-success (fn [community]
+                                    (rf/dispatch [:chat.ui/community-resolved community-id community]))
+                      :on-error   (fn [err]
+                                    (rf/dispatch [:chat.ui/community-failed-to-resolve community-id])
+                                    (log/error {:message
+                                                "Failed to request community info from mailserver"
+                                                :error err}))}]}))
+
+(schema/=> fetch-community
+  [:=>
+   [:catn
+    [:cofx :schema.re-frame/cofx]
+    [:args
+     [:schema [:? :string]]]]
+   [:map
+    [:db map?]
+    [:json-rpc/call :schema.common/rpc-call]]])
+
+(rf/reg-event-fx :chat.ui/fetch-community fetch-community)
+
+(defn spectate-community-successed
+  [{:keys [db]} [{:keys [communities]}]]
+  (when-let [community (first communities)]
+    {:db (-> db
+             (assoc-in [:communities (:id community) :spectated] true)
+             (assoc-in [:communities (:id community) :spectating] false))
+     :fx [[:dispatch [:communities/handle-community community]]
+          [:dispatch [::mailserver/request-messages]]]}))
+
+(rf/reg-event-fx :chat.ui/spectate-community-successed spectate-community-successed)
+
+(defn spectate-community-failed
+  [{:keys [db]} [community-id]]
+  {:db (assoc-in db [:communities community-id :spectating] false)})
+
+(rf/reg-event-fx :chat.ui/spectate-community-failed spectate-community-failed)
+
+(defn spectate-community
+  [{:keys [db]} [community-id]]
+  (let [{:keys [spectated spectating joined]} (get-in db [:communities community-id])]
+    (when (and (not joined) (not spectated) (not spectating))
+      {:db            (assoc-in db [:communities community-id :spectating] true)
+       :json-rpc/call [{:method     "wakuext_spectateCommunity"
+                        :params     [community-id]
+                        :on-success (fn [res]
+                                      (rf/dispatch [:chat.ui/spectate-community-successed res]))
```

Comment:
If we need to dispatch only a re-frame event on success, we can pass just the vector with the event name.

```suggestion
                        :on-success [:chat.ui/spectate-community-successed]
```

The `:json-rpc/call` effects `conj` the response in the vector and dispatch it for us.

https://github.com/status-im/status-mobile/blob/c1a17d9d53e3c4d884631fe27d6a942206000106/src/status_im2/common/json_rpc/events.cljs#L75-L87


P.S. The `successed` needs to be updated to  `success` in `chat.ui/spectate-community-successed` as it's grammatically not right.

Reply:
Thanks, addressed above feedback